Springboot 3.3.1 breaks jkube openshift plugin

### Describe the bug

springboot 3.3.1 comes with a new layertool that breaks jkube openshift plugin. Projects upgrading to springboot 3.3.1 will get the following error when they run oc:build command:

`[ERROR] Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.16.2:build (default-cli) on project layered-jar-3.3.1-bug: Failed to execute the build: Unable to build the image using the OpenShift build service: Assemblies with more than one layer require a proper id for each layer -> [Help 1]
`

The core of the issue is the new layertool is that there is an extra message accompanied by a new line in the output when command `java -Djarmode=layertools -jar xxx.jar list` is run

```
❯ java -Djarmode=layertools -jar target/layered-jar-3.3.1-bug-0.0.1-SNAPSHOT.jar list
Warning: This command is deprecated. Use '-Djarmode=tools list-layers' instead.

dependencies
spring-boot-loader
snapshot-dependencies
application
```

Jkube identifies the empty line as an individual layer without an id and fails with error message provided above. Additionally, JKube identifies any line output in the layertool as a layer. In our case, layer tool has additional lines in the output that are picked as layers erroneously. Here is a sample layertool output on our project:

```
❯ java -Djarmode=layertools -jar payment-service-0.0.1-SNAPSHOT.jar list
Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [ch.qos.logback.classic.spi.LogbackServiceProvider@52a86356]
SLF4J(W): Found provider [org.apache.logging.slf4j.SLF4JServiceProvider@5ce81285]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [ch.qos.logback.classic.spi.LogbackServiceProvider@52a86356]
Warning: This command is deprecated. Use '-Djarmode=tools list-layers' instead.

dependencies
spring-boot-loader
snapshot-dependencies
application
```

in the above case, JKube creates the following 8 layers where as only layer 4-7 are applicable.

1. Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
2. Warning: This command is deprecated. Use '-Djarmode=tools list-layers' instead.
3. 
4. dependencies
5. spring-boot-loader
6. snapshot-dependencies
7. application

### Eclipse JKube version

1.16.2

### Component

OpenShift Maven Plugin

### Apache Maven version

other (please specify in additional context)

### Gradle version

None

### Steps to reproduce

run `mvn package oc:build` on the project in https://github.com/Farzad407/layered-jar-3.3.1-bug repository.

### Expected behavior

image build task to succeed.

### Runtime

OpenShift

### Kubernetes API Server version

1.25.3

### Environment

macOS

### Eclipse JKube Logs

```shell
[INFO] --- oc:1.16.2:build (default-cli) @ layered-jar-3.3.1-bug ---
[INFO] oc: Using OpenShift build with strategy Docker
[INFO] oc: Running generator spring-boot
[INFO] oc: spring-boot: Using Docker image quay.io/jkube/jkube-java:0.0.23 as base / builder
[INFO] oc: Spring Boot layered jar detected
[ERROR] oc: Failed to execute the build [Unable to build the image using the OpenShift build service]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.364 s
[INFO] Finished at: 2024-07-15T15:40:45-04:00
[INFO] ------------------------------------------------------------------------
[WARNING] The requested profile "prodprofile" could not be activated because it does not exist.
[ERROR] Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.16.2:build (default-cli) on project layered-jar-3.3.1-bug: Failed to execute the build: Unable to build the image using the OpenShift build service: Assemblies with more than one layer require a proper id for each layer -> [Help 1]
```


### Sample Reproducer Project

https://github.com/Farzad407/layered-jar-3.3.1-bug

### Additional context

maven version 3.9.6 but it should be reproducible across all maven versions.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -44,6 +44,7 @@ Usage:
 * Fix #3161: JavaExecGenerator should honor %t setting and not unconditionally add `latest` tag
 * Fix #2467: Add support for specifying imagePullSecrets via resource configuration
 * Fix #3220: ImageEnricher#mergeEnvVariables causes error for empty env
+* Fix #3228: Springboot 3.3.1 layertools output format breaks LayeredJarGenerator
 
 _**Note**_:
 - `defaultStorageClass` and `useStorageClassAnnotation` fields have been removed from VolumePermissionEnricher (`jkube-volume-permission`). Users are advised to use these fields from PersistentVolumeClaimStorageClassEnricher (`jkube-persistentvolumeclaim-storageclass`) instead.

--- a/jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/SpringBootLayeredJar.java
+++ b/jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/SpringBootLayeredJar.java
@@ -13,19 +13,20 @@
  */
 package org.eclipse.jkube.springboot;
 
-import lombok.Getter;
 import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.eclipse.jkube.kit.common.ExternalCommand;
 import org.eclipse.jkube.kit.common.KitLogger;
+import org.eclipse.jkube.kit.common.util.Serialization;
 
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import java.util.Properties;
 import java.util.jar.JarFile;
+import java.util.stream.Collectors;
 import java.util.zip.ZipEntry;
 
 public class SpringBootLayeredJar {
@@ -65,10 +66,15 @@ public class SpringBootLayeredJar {
   }
 
   public List<String> listLayers() {
-    final LayerListCommand layerListCommand = new LayerListCommand(kitLogger, layeredJar);
-    try {
-      layerListCommand.execute();
-      return layerListCommand.getLayers();
+    try (JarFile jarFile = new JarFile(layeredJar)) {
+      List<Map<String, List<String>>> layers = Serialization.unmarshal(jarFile.getInputStream(jarFile.getEntry("BOOT-INF/layers.idx")), List.class);
+      if (layers == null) {
+        throw new IOException("Unable to find layers information in BOOT-INF/layers.idx file");
+      }
+
+      return layers.stream()
+          .flatMap(m -> m.keySet().stream())
+          .collect(Collectors.toList());
     } catch (IOException ioException) {
       throw new IllegalStateException("Failure in getting spring boot jar layers information", ioException);
     }
@@ -98,18 +104,4 @@ public class SpringBootLayeredJar {
     }
   }
 
-  @Getter
-  private static class LayerListCommand extends LayerToolsCommand {
-
-    private final List<String> layers;
-    protected LayerListCommand(KitLogger log, File layeredJar) {
-      super(log, null, layeredJar, "list");
-      layers = new ArrayList<>();
-    }
-
-    @Override
-    protected void processLine(String line) {
-      layers.add(line);
-    }
-  }
 }

